contracts/mixnet/src/queries.rs

```diff
@@ -786,6 +840,248 @@ mod tests {
+
+            // page1 should have 2 results on it
+            let page1 = query_mixnode_reverse_delegations_paged(
+                deps.as_ref(),
+                delegation_owner.clone(),
+                None,
+                Option::from(per_page),
+            )
+            .unwrap();
+            assert_eq!(2, page1.delegated_nodes.len());
+
+            mix_reverse_delegations(&mut deps.storage, &delegation_owner)
+                .save("3".as_bytes(), &())
+                .unwrap();
+
+            // page1 still has 2 results
+            let page1 = query_mixnode_reverse_delegations_paged(
+                deps.as_ref(),
+                delegation_owner.clone(),
+                None,
+                Option::from(per_page),
+            )
+            .unwrap();
+            assert_eq!(2, page1.delegated_nodes.len());
+
+            // retrieving the next page should start after the last key on this page
+            let start_after: IdentityKey = String::from("2");
+            let page2 = query_mixnode_reverse_delegations_paged(
+                deps.as_ref(),
+                delegation_owner.clone(),
+                Option::from(start_after),
+                Option::from(per_page),
+            )
+            .unwrap();
+
+            assert_eq!(1, page2.delegated_nodes.len());
+
+            // save another one
+            mix_reverse_delegations(&mut deps.storage, &delegation_owner)
+                .save("4".as_bytes(), &())
+                .unwrap();
+
+            let start_after = String::from("2");
+            let page2 = query_mixnode_reverse_delegations_paged(
+                deps.as_ref(),
+                delegation_owner,
+                Option::from(start_after),
+                Option::from(per_page),
+            )
+            .unwrap();
+
+            // now we have 2 pages, with 2 results on the second page
+            assert_eq!(2, page2.delegated_nodes.len());
+        }
+
+        #[test]
+        fn mix_reverse_deletion_query_returns_current_delegation_value() {
+            let mut deps = helpers::init_contract();
+            let node_identity: IdentityKey = "foo".into();
+            let delegation_owner = Addr::unchecked("bar");
+
+            mix_reverse_delegations(&mut deps.storage, &delegation_owner)
+                .save(node_identity.as_bytes(), &())
+                .unwrap();
+
+            assert_eq!(
+                Ok(()),
+                query_mixnode_reverse_delegation(deps.as_ref(), node_identity, delegation_owner)
+            )
+        }
+
+        #[test]
+        fn mix_reverse_deletion_query_returns_error_if_delegation_doesnt_exist() {
+            let mut deps = helpers::init_contract();
+
+            let node_identity1: IdentityKey = "foo1".into();
+            let node_identity2: IdentityKey = "foo2".into();
+            let delegation_owner1 = Addr::unchecked("bar");
+            let delegation_owner2 = Addr::unchecked("bar2");
+
+            assert_eq!(
```

Comment:
This feels weird, you're asserting correct error type of a helper function you wrote for tests that only returns one error. Are we sure you're testing the correct thing here?

Reply:
You're right, it is pretty weird now that I think of it. The actual important thing here is not the error itself, but the fact that it is an error, and that delegation doesn't exist. Having said that, the correct thing is tested, but in an indirect way, I'll fix it in the next commits.